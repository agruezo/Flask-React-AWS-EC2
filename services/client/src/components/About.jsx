import React from "react";

const About = () => (
  <div>
    <h1 className="title is-1">About</h1>
    <hr />
    <br />
    <h4 className="title is-4">
      Deploying a Flask and React Microservice to AWS EC2
    </h4>
    <h5 className="title is-5">Objectives</h5>
    <h6 className="title is-6">Part 1</h6>
    <ul className="content">
      <li>Create a production Dockerfile that uses multistage Docker builds</li>
      <li>
        Utilize Amazon Elastic Container Registry (ECR) to store built images
      </li>
      <li>Configure CodeBuild to run when code is checked in to GitHub</li>
      <li>
        Run unit and integrations tests and check code for quality and
        formatting issues on CodeBuild
      </li>
    </ul>
    <h6 className="title is-6">Part 2</h6>
    <ul className="content">
      <li>Configure RDS for data persistence</li>
      <li>
        Configure an Application Load Balancer (ALB) along with ECS to run a set
        of microservices
      </li>
      <li>Send container logs to CloudWatch</li>
      <li>
        Update a running container via a zero-downtime deployment strategy to
        not disrupt the current users or application
      </li>
      <li>Use AWS Fargate with ECS to deploy a microservice</li>
      <li>Spin up AWS infrastructure via Terraform</li>
    </ul>
  </div>
);

export default About;
