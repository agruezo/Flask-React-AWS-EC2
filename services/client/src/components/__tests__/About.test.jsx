/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";
import { render, cleanup } from "@testing-library/react";

import About from "../About";

afterEach(cleanup);

it("renders properly", () => {
  const { getByText } = render(<About />);
  expect(
    getByText("Deploying a Flask and React Microservice to AWS EC2")
  ).toHaveClass("title");
});

it("renders", () => {
  const { asFragment } = render(<About />);
  expect(asFragment()).toMatchSnapshot();
});
