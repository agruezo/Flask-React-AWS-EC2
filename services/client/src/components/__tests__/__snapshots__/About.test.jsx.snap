// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="title is-1"
    >
      About
    </h1>
    <hr />
    <br />
    <h4
      class="title is-4"
    >
      Deploying a Flask and React Microservice to AWS EC2
    </h4>
    <h5
      class="title is-5"
    >
      Objectives
    </h5>
    <h6
      class="title is-6"
    >
      Part 1
    </h6>
    <ul
      class="content"
    >
      <li>
        Create a production Dockerfile that uses multistage Docker builds
      </li>
      <li>
        Utilize Amazon Elastic Container Registry (ECR) to store built images
      </li>
      <li>
        Configure CodeBuild to run when code is checked in to GitHub
      </li>
      <li>
        Run unit and integrations tests and check code for quality and formatting issues on CodeBuild
      </li>
    </ul>
    <h6
      class="title is-6"
    >
      Part 2
    </h6>
    <ul
      class="content"
    >
      <li>
        Configure RDS for data persistence
      </li>
      <li>
        Configure an Application Load Balancer (ALB) along with ECS to run a set of microservices
      </li>
      <li>
        Send container logs to CloudWatch
      </li>
      <li>
        Update a running container via a zero-downtime deployment strategy to not disrupt the current users or application
      </li>
      <li>
        Use AWS Fargate with ECS to deploy a microservice
      </li>
      <li>
        Spin up AWS infrastructure via Terraform
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
